### DISCUSSION TITLE
feature request: mouseover preview for the 'add' function, undo/redo

### DISCUSSION THREAD

**Discussion by raminrahni** [WebSVG/voronoi#21] [Ideas]
Thank you so much for creating and sharing this wonderful tool! It's incredibly useful. Are there any plans to implement a preview feature for the 'add' function? It would be super helpful to know where the cell is going to be cut before clicking, especially in the absence of an undo function (another feature that would be extremely useful). Thanks!

**Comment by wassfila** [comment]
Hi,
thank you for your encouragement. I understand what you're asking for and it makes total sense. I keep this ticket open. Although  I no longer spend developments efforts on this project, I will think about this feature.
In the meanwhile, it is important to notice that this preview can be reproduced by first drooping the cell wherever, and then switch to the "move" mode and move that same exact cell, if happy with it, then keep it, and if not, you would have to switch to delete mode and remove that same last cell. Also the undo, can be manually done by removing the cells you do not like in the back order.
Now, of course, these two features can be great additions and I will write some notes here on to how to perform them. If you or anyone is willing to give it a try, I am willing even to spend more effort on explaining the framework to someone than it would take to do it myself, as I see that more beneficial.

**Reply by raminrahni** [reply]
Thanks for the quick reply and for the workaround suggestions + implementation notes!

**Comment by wassfila** [comment]
As this relates to a direct code change request, I moved it to an issue #22 but it is fine to keep this thread for discussion and the other one optionally for more specific code related changes.
